src/fluree/db/nameservice/storage.cljc

```diff
@@ -127,20 +184,34 @@
       ;; Use recursive listing to support ledger names with '/' characters
       (if (satisfies? storage/RecursiveListableStore store)
         (if-let [list-paths-result (storage/list-paths-recursive store const/ns-version)]
-          (loop [remaining-paths (<? list-paths-result)
-                 records         []]
-            (if-let [path (first remaining-paths)]
-              (let [file-content (<? (storage/read-bytes store path))]
-                (if file-content
-                  (let [content-str (if (string? file-content)
-                                      file-content
-                                      #?(:clj (let [^bytes bytes-content file-content]
-                                                (String. bytes-content "UTF-8"))
-                                         :cljs (js/String.fromCharCode.apply nil file-content)))
-                        record      (json/parse content-str false)]
-                    (recur (rest remaining-paths) (conj records record)))
-                  (recur (rest remaining-paths) records)))
-              records))
+          (let [all-paths (<? list-paths-result)
+                ;; Filter out .index.json files - they're supplementary and will be merged
+                main-paths (filterv #(not (str/ends-with? % ".index.json")) all-paths)]
+            (loop [remaining-paths main-paths
+                   records         []]
+              (if-let [path (first remaining-paths)]
+                (let [file-content (<? (storage/read-bytes store path))]
+                  (if file-content
```

Comment:
This should be converted to `if-some`

Reply:
Addressed in bf7812d745e17b7ce66144d2bf84676706e40ee8